jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1147,6 +1147,15 @@ public static ClassLoader getSystemClassLoader () {
 	return sysLoader;
 }
 
+/*[IF JAVA_SPEC_VERSION >= 19]*/
+/*
+ *  Returns the system class loader without a security check.
+*/
+static ClassLoader internalGetSystemClassLoader() {
+	return applicationClassLoader;
```

Comment:
We don't need that. The code in getSystemClassLoader() only runs when initSystemClassLoader is true, which is set in completeInitialization(). `applicationClassLoader` is first set in initializeClassLoaders(), which is called from Thread.initialize() when booting. If `applicationClassLoader` is null, it's very early in bootstrap and it would be appropriate to return null since only the bootstrap loader is available. Trying to change the bootstrap could cause a failure to startup and those are hard to diagnose. The VM bootstrap sequence would need to change to call Class.forName() with the intention to load a non-bootstrap class for any change to be necessary.

Reply:
I forgot to mention that after ClassLoader.completeInitialization() is called, Thread.threadCompleteInitialization() is called which calls ClassLoader.getSystemClassLoader(). i.e. the bootstrap ensures the code in getSystemClassLoader() for `initSystemClassLoader ` runs at the appropriate time.

Reply:
That seems reasonable; thanks for the explanation.